device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubAuthenticationProvider.java

```diff
@@ -25,100 +27,35 @@
     protected String moduleId;
 
     protected IotHubSSLContext iotHubSSLContext;
-    protected boolean sslContextNeedsUpdate;
-    protected String iotHubTrustedCert;
-    protected String pathToIotHubTrustedCert;
```

Comment:
SSLContext is now set at client construction time, so there is no case where the SSLContext needs updating, nor is there a means to set a new trusted later.

Reply:
Can you elaborate on why (if you know) we postponed creating the SSL context and had to have all this logic in place?

Reply:
There are some Edgehub related scenarios where we needed to trust a certificate other than the base IoT hub cert, so we added these features to enable those scenarios. At this point though, it is reasonable to expect a user to know what the SSLContext looks like at client construction time